Neos.Flow/Classes/Log/Utility/LogEnvironment.php

```diff
@@ -61,9 +54,21 @@ public static function fromMethodName(string $methodName): array
     }
 
     /**
-     * @param string $className
-     * @return string
+     * Returns an array containing the log environment variables
+     * under the key FLOW_LOG_ENVIRONMENT to be set as part of the
+     * additional data in an log method call.
      */
+    public static function fromClassAndMethodName(string $className, string $methodName): array
```

Comment:
What if we also allow passing an object instead of the class name? That way you could just pass `$this` in most cases.

```php
public static function fromClassAndMethodName(string|object $class, string $methodName): array
{
    $className = is_object($class) ? get_class($object) : $className;
...
```

Reply:
Does that make it easier to use? You can even use `$this::class` if you like…